src/c++/perf_analyzer/inference_profiler.cc

```diff
@@ -1517,7 +1517,8 @@ InferenceProfiler::DetermineStatsModelVersion(
     }
   }
 
-  if (*status_model_version == -1) {
+  if (*status_model_version == -1 &&
+      (parser_ == nullptr || !parser_->TopLevelResponseCachingEnabled())) {
```

Comment:
What does this combination represent? Can we assign a meaningful variable name instead?
```cpp
bool meaningful_name = (parser_ == nullptr || !parser_->TopLevelResponseCachingEnabled());
if (*status_model_version == -1 && meaningful_name) {
  ...
}
```